app/coffee/ChannelManager.coffee

```diff
@@ -31,7 +31,13 @@ module.exports = ChannelManager =
             clientChannelMap.set(channel, subscribePromise)
             logger.log {channel}, "subscribed to new channel"
             metrics.inc "subscribe.#{baseChannel}"
-            return subscribePromise
+            return new Promise (resolve, reject) ->
+                subscribePromise.then(resolve)
+                subscribePromise.catch (err) ->
+                    metrics.inc "subscribe.failed.#{baseChannel}"
+                    # clear state
+                    clientChannelMap.delete(channel)
+                    reject(err)
```

Comment:
could we do this without creating a new promise?

```
subscribePromise.catch (err) -> 
    # handle failure case
    metrics.inc ..
    clientChannelMap.delete(channel)
return subscribePromise
```

Reply:
Good idea, done in https://github.com/overleaf/real-time/pull/130/commits/e8d18b271a73ecf6559a592046eeebbf8e8c7017